src/ripple/overlay/impl/Message.cpp

```diff
@@ -36,23 +40,86 @@ Message::Message (::google::protobuf::Message const& message, int type)
 
     assert (messageBytes != 0);
 
-    /** Number of bytes in a message header. */
-    std::size_t constexpr headerBytes = 6;
-
     mBuffer.resize (headerBytes + messageBytes);
 
-    auto ptr = mBuffer.data();
+    setHeader(mBuffer.data(), messageBytes, type, Compressed::Off);
 
-    *ptr++ = static_cast<std::uint8_t>((messageBytes >> 24) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t>((messageBytes >> 16) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t>((messageBytes >> 8) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t>(messageBytes & 0xFF);
+    if (messageBytes != 0)
+        message.SerializeToArray(mBuffer.data() + headerBytes, messageBytes);
+}
 
-    *ptr++ = static_cast<std::uint8_t>((type >> 8) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t> (type & 0xFF);
+void
+Message::compress()
+{
+    auto const messageBytes = mBuffer.size () - headerBytes;
 
-    if (messageBytes != 0)
-        message.SerializeToArray(ptr, messageBytes);
+    auto type = getType(mBuffer.data());
+
+    bool const compressible = [&]{
+        if (messageBytes <= 70)
+            return false;
+        switch(type)
+        {
+            case protocol::mtMANIFESTS:
```

Comment:
Majority of the messages are not compressible (88% in the sample). The filtering is desirable because both CPU and memory (have to allocate compressed message buffer) will be used. Perhaps we could mitigate maintainability concern by adding compiler warning when a case is missing in the switch?